internal/service/employeeaccessgrant/model_test.go

```diff
@@ -0,0 +1,80 @@
+package employeeaccessgrant_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/employeeaccessgrant"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+func TestNewTFModel(t *testing.T) {
+	testCases := map[string]struct {
+		apiResp         *admin.EmployeeAccessGrant
+		expectedTFModel *employeeaccessgrant.TFModel
+		projectID       string
+		clusterName     string
+	}{
+		"valid": {
+			projectID:   "projectID",
+			clusterName: "clusterName",
+			apiResp: &admin.EmployeeAccessGrant{
+				GrantType:      "grantType",
+				ExpirationTime: time.Date(2024, 10, 13, 0, 0, 0, 0, time.UTC),
+			},
+			expectedTFModel: &employeeaccessgrant.TFModel{
+				ProjectID:      types.StringValue("projectID"),
+				ClusterName:    types.StringValue("clusterName"),
+				GrantType:      types.StringValue("grantType"),
+				ExpirationTime: types.StringValue("2024-10-13T00:00:00Z"),
+			},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			tfModel := employeeaccessgrant.NewTFModel(tc.projectID, tc.clusterName, tc.apiResp)
+			assert.Equal(t, tc.expectedTFModel, tfModel)
+		})
+	}
+}
+
+func TestNewAtlasReq(t *testing.T) {
+	testCases := map[string]struct {
+		tfModel        *employeeaccessgrant.TFModel
+		expectedReq    *admin.EmployeeAccessGrant
+		expectedHasErr bool
+	}{
+		"valid": {
+			tfModel: &employeeaccessgrant.TFModel{
+				ProjectID:      types.StringValue("projectID"),
+				ClusterName:    types.StringValue("clusterName"),
+				GrantType:      types.StringValue("grantType"),
+				ExpirationTime: types.StringValue("2024-10-13T00:00:00Z"),
+			},
+			expectedReq: &admin.EmployeeAccessGrant{
+				GrantType:      "grantType",
+				ExpirationTime: time.Date(2024, 10, 13, 0, 0, 0, 0, time.UTC),
+			},
+		},
+		"invalid expiration time": {
+			tfModel: &employeeaccessgrant.TFModel{
+				ProjectID:      types.StringValue("projectID"),
+				ClusterName:    types.StringValue("clusterName"),
+				GrantType:      types.StringValue("grantType"),
+				ExpirationTime: types.StringValue("invalid_time"),
+			},
+			expectedHasErr: true,
```

Comment:
what about `expectErrString` and use an `assert.Contains` instead?

Reply:
thanks, added in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2573/commits/646b57b32c0bedddaf7a4cdeb2776cdbc42bac35